src/main/java/org/benetech/mathshare/service/ProblemService.java

```diff
@@ -0,0 +1,35 @@
+package org.benetech.mathshare.service;
+
+import org.benetech.mathshare.converters.UrlCodeConverter;
+import org.benetech.mathshare.mappers.ProblemMapper;
+import org.benetech.mathshare.model.dto.ProblemDTO;
+import org.benetech.mathshare.model.dto.ProblemSetDTO;
+import org.benetech.mathshare.model.entity.ProblemSetRevision;
+import org.benetech.mathshare.repository.ProblemRepository;
+import org.benetech.mathshare.repository.ProblemSetRevisionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProblemService {
+
+    @Autowired
+    private ProblemSetRevisionRepository problemSetRevisionRepository;
+
+    @Autowired
+    private ProblemRepository problemRepository;
+
+    public ProblemSetDTO findProblemsByUrlCode(String code) throws IllegalArgumentException {
```

Comment:
You should add @Transactional(readOnly = true) to use efficient, read-only transactions

Reply:
Ok